// @superfluid-finance/ethereum-contracts/build/typechain-ethers-v5/factories/contracts/mocks/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UUPSProxiableMock__factory = exports.SuperfluidDestructorMock__factory = exports.SuperAppBaseFlowTester__factory = exports.StreamRedirector__factory = exports.MultiFlowTesterApp__factory = exports.MockSmartWallet__factory = exports.IDASuperAppTester__factory = exports.ForwarderMock__factory = exports.FakeSuperfluidMock__factory = exports.CallUtilsTester__factory = exports.CallUtilsMock__factory = exports.AgreementMock__factory = exports.superfluidMockSol = exports.superfluidGovernanceIiMockSol = exports.superTokenMockSol = exports.superTokenLibraryV1MockSol = exports.superTokenFactoryMockSol = exports.superAppMocksSol = exports.idAv1LibraryMockSol = exports.erc777SenderRecipientMockSol = exports.customSuperTokenMockSol = exports.cfAv1NftUpgradabilityMockSol = exports.cfAv1NftMockSol = exports.cfaLibraryMockSol = exports.cfaAppMocksSol = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.cfaAppMocksSol = __importStar(require("./CFAAppMocks.sol"));
exports.cfaLibraryMockSol = __importStar(require("./CFALibraryMock.sol"));
exports.cfAv1NftMockSol = __importStar(require("./CFAv1NFTMock.sol"));
exports.cfAv1NftUpgradabilityMockSol = __importStar(require("./CFAv1NFTUpgradabilityMock.sol"));
exports.customSuperTokenMockSol = __importStar(require("./CustomSuperTokenMock.sol"));
exports.erc777SenderRecipientMockSol = __importStar(require("./ERC777SenderRecipientMock.sol"));
exports.idAv1LibraryMockSol = __importStar(require("./IDAv1LibraryMock.sol"));
exports.superAppMocksSol = __importStar(require("./SuperAppMocks.sol"));
exports.superTokenFactoryMockSol = __importStar(require("./SuperTokenFactoryMock.sol"));
exports.superTokenLibraryV1MockSol = __importStar(require("./SuperTokenLibraryV1Mock.sol"));
exports.superTokenMockSol = __importStar(require("./SuperTokenMock.sol"));
exports.superfluidGovernanceIiMockSol = __importStar(require("./SuperfluidGovernanceIIMock.sol"));
exports.superfluidMockSol = __importStar(require("./SuperfluidMock.sol"));
var AgreementMock__factory_1 = require("./AgreementMock__factory");
Object.defineProperty(exports, "AgreementMock__factory", { enumerable: true, get: function () { return AgreementMock__factory_1.AgreementMock__factory; } });
var CallUtilsMock__factory_1 = require("./CallUtilsMock__factory");
Object.defineProperty(exports, "CallUtilsMock__factory", { enumerable: true, get: function () { return CallUtilsMock__factory_1.CallUtilsMock__factory; } });
var CallUtilsTester__factory_1 = require("./CallUtilsTester__factory");
Object.defineProperty(exports, "CallUtilsTester__factory", { enumerable: true, get: function () { return CallUtilsTester__factory_1.CallUtilsTester__factory; } });
var FakeSuperfluidMock__factory_1 = require("./FakeSuperfluidMock__factory");
Object.defineProperty(exports, "FakeSuperfluidMock__factory", { enumerable: true, get: function () { return FakeSuperfluidMock__factory_1.FakeSuperfluidMock__factory; } });
var ForwarderMock__factory_1 = require("./ForwarderMock__factory");
Object.defineProperty(exports, "ForwarderMock__factory", { enumerable: true, get: function () { return ForwarderMock__factory_1.ForwarderMock__factory; } });
var IDASuperAppTester__factory_1 = require("./IDASuperAppTester__factory");
Object.defineProperty(exports, "IDASuperAppTester__factory", { enumerable: true, get: function () { return IDASuperAppTester__factory_1.IDASuperAppTester__factory; } });
var MockSmartWallet__factory_1 = require("./MockSmartWallet__factory");
Object.defineProperty(exports, "MockSmartWallet__factory", { enumerable: true, get: function () { return MockSmartWallet__factory_1.MockSmartWallet__factory; } });
var MultiFlowTesterApp__factory_1 = require("./MultiFlowTesterApp__factory");
Object.defineProperty(exports, "MultiFlowTesterApp__factory", { enumerable: true, get: function () { return MultiFlowTesterApp__factory_1.MultiFlowTesterApp__factory; } });
var StreamRedirector__factory_1 = require("./StreamRedirector__factory");
Object.defineProperty(exports, "StreamRedirector__factory", { enumerable: true, get: function () { return StreamRedirector__factory_1.StreamRedirector__factory; } });
var SuperAppBaseFlowTester__factory_1 = require("./SuperAppBaseFlowTester__factory");
Object.defineProperty(exports, "SuperAppBaseFlowTester__factory", { enumerable: true, get: function () { return SuperAppBaseFlowTester__factory_1.SuperAppBaseFlowTester__factory; } });
var SuperfluidDestructorMock__factory_1 = require("./SuperfluidDestructorMock__factory");
Object.defineProperty(exports, "SuperfluidDestructorMock__factory", { enumerable: true, get: function () { return SuperfluidDestructorMock__factory_1.SuperfluidDestructorMock__factory; } });
var UUPSProxiableMock__factory_1 = require("./UUPSProxiableMock__factory");
Object.defineProperty(exports, "UUPSProxiableMock__factory", { enumerable: true, get: function () { return UUPSProxiableMock__factory_1.UUPSProxiableMock__factory; } });
