// @superfluid-finance/ethereum-contracts/build/typechain-ethers-v5/factories/contracts/utils/TOGA.sol/ITOGAv3__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ITOGAv3__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "additionalBond",
                type: "uint256",
            },
        ],
        name: "BondIncreased",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                indexed: false,
                internalType: "int96",
                name: "exitRate",
                type: "int96",
            },
        ],
        name: "ExitRateChanged",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                indexed: false,
                internalType: "address",
                name: "pic",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "bond",
                type: "uint256",
            },
            {
                indexed: false,
                internalType: "int96",
                name: "exitRate",
                type: "int96",
            },
        ],
        name: "NewPIC",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                internalType: "int96",
                name: "newExitRate",
                type: "int96",
            },
        ],
        name: "changeExitRate",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
        ],
        name: "getCurrentPIC",
        outputs: [
            {
                internalType: "address",
                name: "pic",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
        ],
        name: "getCurrentPICInfo",
        outputs: [
            {
                internalType: "address",
                name: "pic",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "bond",
                type: "uint256",
            },
            {
                internalType: "int96",
                name: "exitRate",
                type: "int96",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "bondAmount",
                type: "uint256",
            },
        ],
        name: "getDefaultExitRateFor",
        outputs: [
            {
                internalType: "int96",
                name: "exitRate",
                type: "int96",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISuperToken",
                name: "token",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "bondAmount",
                type: "uint256",
            },
        ],
        name: "getMaxExitRateFor",
        outputs: [
            {
                internalType: "int96",
                name: "exitRate",
                type: "int96",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class ITOGAv3__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ITOGAv3__factory = ITOGAv3__factory;
ITOGAv3__factory.abi = _abi;
